app/code/Magento/CatalogInventory/Test/Unit/Observer/InvalidatePriceIndexUponConfigChangeObserverTest.php

```diff
@@ -0,0 +1,118 @@
+use Magento\Framework\Indexer\IndexerInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class InvalidatePriceIndexUponConfigChangeObserverTest
+ *
+ * Testing invalidating product price index onn config changing
+ */
+class InvalidatePriceIndexUponConfigChangeObserverTest extends TestCase
+{
+    /**
+     * @var InvalidatePriceIndexUponConfigChangeObserver
+     */
+    private $observer;
+
+    /**
+     * @var Processor|MockObject
+     */
+    private $priceIndexProcessorMock;
+
+    /**
+     * @var Observer|MockObject
+     */
+    private $observerMock;
+
+    /**
+     * @var Event|MockObject
+     */
+    private $eventMock;
+
+    /**
+     * @var IndexerInterface|MockObject
+     */
+    private $indexerMock;
+
+    /**
+     * Set Up
+     */
+    public function setUp()
+    {
+        $objectManager = new ObjectManager($this);
+        $this->priceIndexProcessorMock = $this->createMock(Processor::class);
+        $this->indexerMock = $this->getMockBuilder(IndexerInterface::class)
+            ->getMockForAbstractClass();
+        $this->observerMock = $this->createMock(Observer::class);
+        $this->eventMock = $this->getMockBuilder(Event::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getChangedPaths'])
+            ->getMock();
+
+        $this->observer = $objectManager->getObject(
+            InvalidatePriceIndexUponConfigChangeObserver::class,
+            [
+                'priceIndexProcessor' => $this->priceIndexProcessorMock
+            ]
+        );
+    }
+
+    /**
+     * Testing invalidating product price index on catalog inventory config changes
+     */
+    public function testInvalidatingPriceOnChangingOutOfStockConfig()
+    {
+        $changedPaths = [Configuration::XML_PATH_SHOW_OUT_OF_STOCK];
+
+        $this->eventMock->expects($this->once())
+            ->method('getChangedPaths')
+            ->willReturn($changedPaths);
+        $this->observerMock->expects($this->once())
+            ->method('getEvent')
+            ->willReturn($this->eventMock);
+        $this->indexerMock->expects($this->once())
+            ->method('invalidate');
+        $this->priceIndexProcessorMock->expects($this->once())
+            ->method('getIndexer')
+            ->willReturn($this->indexerMock);
+
+        $this->observer->execute($this->observerMock);
+    }
+
+    /**
+     * Testing invalidating product price index on changing any other config
+     */
+    public function testInvalidatingPriceOnChangingAnyOtherConfig()
+    {
+        $changedPaths = [Configuration::XML_PATH_ITEM_AUTO_RETURN];
+
+        $this->eventMock->expects($this->once())
+            ->method('getChangedPaths')
+            ->willReturn($changedPaths);
+        $this->observerMock->expects($this->once())
+            ->method('getEvent')
+            ->willReturn($this->eventMock);
```

Comment:
I'm not sure if the expecting single execution of getEvent is critical.
When implementation is changed and extra behaviour is added, then this test would fail - even if the functionality covered by tour test has not changed.